test/test_sparse.py

```diff
@@ -974,20 +974,95 @@ def test_factory_size_check(self):
         values = self.ValueTensor([.5, .5])
         sizes = torch.Size([2, 3])
         with self.assertRaisesRegex(RuntimeError, "sizes is inconsistent with indices"):
-            self.SparseTensor(indices, values, sizes)
+            torch.sparse_coo_tensor(indices, values, sizes)
 
         indices = self.IndexTensor([[1, 2], [0, 2]])
         values = self.ValueTensor([[1, 1, 1], [1, 1, 1]])
         sizes = torch.Size([3, 3, 2])
-        with self.assertRaisesRegex(RuntimeError, "values and sizes are inconsistent"):
-            self.SparseTensor(indices, values, sizes)
+        with self.assertRaisesRegex(RuntimeError, "values has incorrect size"):
+            torch.sparse_coo_tensor(indices, values, sizes)
+
+    def test_factory_default(self):
+        tensor = self.SparseTensor()
+        expected_indices = self.IndexTensor(1, 0)
+        expected_size = torch.Size([0])
+        self.assertEqual(tensor._indices(), expected_indices)
+        self.assertEqual(tensor.shape, expected_size)
 
     def test_factory_empty_indices(self):
         device = 'cuda' if self.is_cuda else 'cpu'
-        tensor = torch.sparse_coo_tensor([], [], torch.Size([]), device=device)
-        expected_indices = torch.tensor([], dtype=torch.long, device=device)
+        tensor = self.SparseTensor()
+        expected_indices = torch.empty((1, 0), dtype=torch.long, device=device)
+        self.assertEqual(tensor._indices(), expected_indices)
+
+        tensor = torch.sparse_coo_tensor(torch.Size([2, 0]), device=device)
+        expected_indices = torch.empty((2, 0), dtype=torch.long, device=device)
+        self.assertEqual(tensor._indices(), expected_indices)
+
+        tensor = torch.sparse_coo_tensor(torch.Size([2, 2, 0]), device=device)
+        expected_indices = torch.empty((3, 0), dtype=torch.long, device=device)
         self.assertEqual(tensor._indices(), expected_indices)
 
+        tensor = torch.sparse_coo_tensor(torch.Size([2, 2, 0, 0]), device=device)
+        expected_indices = torch.empty((4, 0), dtype=torch.long, device=device)
+        self.assertEqual(tensor._indices(), expected_indices)
+
+    def test_factory_nnz(self):
+        indices = self.IndexTensor([[0]])  # (sparseDims, nnz): (1, 1)
+        values = self.ValueTensor([[1, 1], [1, 1]])  # (nnz, ...): (2, 2)
+        sizes = torch.Size([2, 2])
+        with self.assertRaisesRegex(RuntimeError, "indices and values must have same nnz"):
+            torch.sparse_coo_tensor(indices, values, sizes)
+
+    def test_factory_zero_nnz(self):
```

Comment:
nit: call this test_factory_nnz_zero so it's easier to search for, alphabetical order, etc.